torch/_dynamo/convert_frame.py

```diff
@@ -386,21 +387,23 @@ def format_guard_failures(code):
             },
         )
 
-        return _compile(
-            frame.f_code,
-            frame.f_globals,
-            frame.f_locals,
-            frame.f_builtins,
-            compiler_fn,
-            one_graph,
-            export,
-            export_constraints,
-            hooks,
-            cache_size,
-            frame,
-            frame_state=frame_state,
-            compile_id=compile_id,
-        )
+        with config.patch(patch_config_if_changed()):
```

Comment:
since compiling is already pretty expensive, it doesn't seem like a huge deal to just unconditionally patch in all the config all the time